### PR TITLE
Using more often names of universes in universe inconsistency message, if possible

### PR DESCRIPTION
**Pull request by herbelin**
The PR tries to use the universe names found in the universe state when reporting about a universe inconsistency (though it is not always possible, the way the code is organized).

Example:
```coq
Set Universe Polymorphism.
Definition g@{a b} := Type@{a} : Type@{b}.
Fail Definition h@{a} := g@{a a}.
(* Before: Universe inconsistency. Cannot enforce Top.3 < Top.3 because Top.3 = Top.3.
    After: Universe inconsistency. Cannot enforce a < a because a = a.
```
- [x] Added / updated **test-suite**.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
I thought about doing something like this but having to chase all the places where ugraph's univ inconsistency is caught seemed too footgunny (indeed the test suite is broken probably for this reason).

**Comment by herbelin** [comment]
> having to chase all the places where ugraph's univ inconsistency is caught seemed too footgunny

You're right. So, what could we do?

**Comment by SkySkimmer** [comment]
We could put the printer data in the ugraph exn, eg
~~~ocaml
(* in ugraph *)
exception UniverseInconsistency of (Level.t -> Pp.t) option * univ_inconsistency

(* in ustate *)
let merge_constraints uctx cstrs g =
  try UGraph.merge_constraints cstrs g
  with UGraph.UniverseInconsistency (None,i) ->
    raise (UniverseInconsistency (Some (pp_level uctx), i))
~~~
(in practice we also want the qvar printer)

**Comment by herbelin** [comment]
Ok, why not.

**Comment by herbelin** [comment]
New version passing the printers, even if still a bit convoluted...

Incidentally, are there some plans to move the names to the kernel (like it is done for `fun`/`forall`/`let` binders)?

**Comment by SkySkimmer** [comment]
>    having to chase all the places where ugraph's univ inconsistency is caught seemed too footgunny

still the case AFAICT

**Comment by herbelin** [comment]
> still the case AFAICT

Was an incomplete push. Retried.

@coqbot run full ci

File: kernel/uGraph.ml [hunk 1]
```diff
@@ -39,7 +39,8 @@ type explanation =
   | Path of path_explanation
   | Other of Pp.t
 
-type univ_inconsistency = constraint_type * Sorts.t * Sorts.t * explanation option
+type univ_variable_printers = (Sorts.QVar.t -> Pp.t) * (Level.t -> Pp.t)
```
Comment (SkySkimmer) [inline kernel/uGraph.ml:42]: We may want to use this type more often instead of passing 2 arguments but not in this PR

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
Shouldn't have merged, this breaks coq lsp

File: kernel/uGraph.mli [hunk 2]
```diff
@@ -50,7 +50,8 @@ type explanation =
   | Path of path_explanation
   | Other of Pp.t
 
-type univ_inconsistency = constraint_type * Sorts.t * Sorts.t * explanation option
+type univ_variable_printers = (Sorts.QVar.t -> Pp.t) * (Level.t -> Pp.t)
```
Comment (ejgallego) [inline kernel/uGraph.mli:53]: Adding functions on error messages is gonna be problematic for IDEs that may want to seralize this stuff for external comsumption.

Comment (herbelin) [reply kernel/uGraph.mli:53]: Ok, so, something still to be done.

**Pull request merged**